File: controller/app/user/form/UserEditForm.kt
package form

import cn.dairo.dfs.dao.UserDao
import cn.dairo.dfs.extension.bean
import jakarta.validation.constraints.AssertTrue
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

type UserEditForm struct{

    /** 主键 **/
    id int64

    /** 用户名 **/
    @Length(min = 2, max = 32)
    @NotBlank
    name string

    @AssertTrue(message = "用户名已经存在")
    fun isName(): Boolean {
        this.name ?: return true
        val existsUser = UserDao::class.bean.selectByName(this.name!!)
        if (this.id == null) {//创建用户时
            if (existsUser != null) {
                return false
            }
        } else {
            if (existsUser != null && existsUser.id != this.id) {
                return false
            }
        }
        return true
    }

    /** 用户电子邮箱 **/
    @Email
    email string

    /** 用户状态 **/
    state int

    /** 创建日期 **/
    date string

    /** 密码 **/
    pwd string

    @AssertTrue(message = "密码必填")
    fun isPwd(): Boolean {
        if (id == null && pwd.isNullOrBlank()) {//创建用户时密码必填
            return false
        }
        return true
    }
}

File: controller/app/user/form/UserEditForm.go
package form

import (
	"DairoDFS/dao/UserDao"
)

type UserEditForm struct {

	/** 主键 **/
	Id *int64 `json:"id"`

	/** 用户名 **/
	Name *string `json:"name" validate:"required,min=2,max=32"`

	/** 用户电子邮箱 **/
	Email *string `json:"email" validate:"email"`

	/** 用户状态 **/
	State *int `json:"state"`

	/** 创建日期 **/
	Date *string `json:"date"`

	/** 密码 **/
	Pwd *string `json:"pwd"`
}

func (mine *UserEditForm) IsName() *string {
	msg := "用户名已经存在"
	existsUser := UserDao.SelectByName(*mine.Name)
	if mine.Id == nil { //创建用户时
		if existsUser != nil {
			return &msg
		}
	} else {
		if existsUser != nil && *existsUser.Id != *mine.Id {
			return &msg
		}
	}
	return nil
}

func (mine *UserEditForm) IsPwd() *string {
	msg := "密码必填"
	if mine.Id == nil && mine.Pwd == nil { //创建用户时密码必填
		return &msg
	}
	return nil
}
